pkg/operator/config_reloader.go

```diff
@@ -179,10 +179,13 @@ func ImagePullPolicy(imagePullPolicy v1.PullPolicy) ReloaderOption {
 	}
 }
 
-// WithNodeNameEnv sets the withNodeNameEnv option for the config-reloader container.
-func WithNodeNameEnv() ReloaderOption {
+// DaemonSet sets the options that only work for DaemonSet mode.
+// Currently we set SHARD env equal to 1, eventhough DaemonSet doesn't use this env.
+// TODO: Remove SHARD env for DaemonSet mode.
+func DaemonSet() ReloaderOption {
 	return func(c *ConfigReloader) {
 		c.withNodeNameEnv = true
+		c.shard = ptr.To(int32(1))
```

Comment:
This needs to be zero because the shards are 0-indexed (otherwise all targets are dropped).
```suggestion
		c.shard = ptr.To(int32(0))
```